portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,8 +62,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String username = request.getParameter("username");
     String message = request.getParameter("comment-or-question");
-    messages.put(username, message);
-    // messages.add(username);
+    String maxNumCommentsString = request.getParameter("max-num-comments");
+    int maxNumComments;
+    try {
+      maxNumComments = Integer.parseInt(maxNumCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.format("Could not convert to int: %d%n", maxNumCommentsString);
+      System.err.format("Using default value of %d.%n", DEFAULT_MAX_NUM_COMMENTS);
+      maxNumComments = DEFAULT_MAX_NUM_COMMENTS;
+    }
+    request.getSession().setAttribute("maxNumComments", maxNumComments);
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("username", username);
+    commentEntity.setProperty("message", message);
+    commentEntity.setProperty("timestamp", System.currentTimeMillis());
```

Comment:
One notable omission here is that the comment is never added to the datastore. Do you want to add that here?

Reply:
That would be useful for the function of the website, so I'll add it.